internal/k8s/controller.go

```diff
@@ -2324,11 +2325,21 @@ func (lbc *LoadBalancerController) createTransportServer(transportServer *conf_v
 
 		endps := getIPAddressesFromEndpoints(podEndps)
 		endpoints[endpointsKey] = endps
+
+		if lbc.isNginxPlus || lbc.isLatencyMetricsEnabled {
```

Comment:
Should this ensure that prometheus is enabled too?

Reply:
@Dean-Coakley you're right. But there are other similar checks in `createIngressEx` and `createVirtualServerEx`. Better to change in those places as well (the change is small, ok to do it along the way)